Sources/SmokeHTTPClient/HTTPOperationsClient +executeAsyncWithOutput.swift

```diff
@@ -79,22 +79,57 @@ public extension HTTPOperationsClient {
             completion: @escaping (Result<OutputType, HTTPClientError>) -> (),
             asyncResponseInvocationStrategy: InvocationStrategyType,
             invocationContext: HTTPClientInvocationContext<InvocationReportingType, HandlerDelegateType>) throws -> EventLoopFuture<HTTPClient.Response>
-            where InputType: HTTPRequestInputProtocol, InvocationStrategyType: AsyncResponseInvocationStrategy,
+    where InputType: HTTPRequestInputProtocol, InvocationStrategyType: AsyncResponseInvocationStrategy,
         InvocationStrategyType.OutputType == Result<OutputType, HTTPClientError>,
         OutputType: HTTPResponseOutputProtocol {
-            let endpoint = getEndpoint(endpointOverride: endpointOverride, path: endpointPath)
+            let requestComponents = try clientDelegate.encodeInputAndQueryString(
+                input: input,
+                httpPath: endpointPath,
+                invocationReporting: invocationContext.reporting)
+            let endpoint = getEndpoint(endpointOverride: endpointOverride, path: requestComponents.pathWithQuery)
             let wrappingInvocationContext = invocationContext.withOutgoingDecoratedLogger(endpoint: endpoint, outgoingOperation: operation)
             
-            return try executeAsyncWithOutputWithWrappedInvocationContext(
+            return try executeAsyncWithOutput(
                 endpointOverride: endpointOverride,
-                endpointPath: endpointPath,
+                requestComponents: requestComponents,
                 httpMethod: httpMethod,
-                input: input,
                 completion: completion,
                 asyncResponseInvocationStrategy: asyncResponseInvocationStrategy,
                 invocationContext: wrappingInvocationContext)
     }
 
+    /**
+     Submits a request that will return a response body to this client asynchronously.
+
+     - Parameters:
+         - requestComponents: The request components for this request.
+         - httpMethod: The http method to use for this request.
+         - input: the input body data to send with this request.
+         - completion: Completion handler called with the response body or any error.
+         - asyncResponseInvocationStrategy: The invocation strategy for the response from this request.
+         - invocationContext: context to use for this invocation.
+     */
+    internal func executeAsyncWithOutput<OutputType, InvocationStrategyType,
```

Comment:
Is there a reason this function is needed? It seems just to delegate to `executeAsyncWithOutputWithWrappedInvocationContext`.